### ISSUE TITLE
Adding one to one relationship disrupts MyModel.find( {criteria:'abc'} )

### ISSUE THREAD

**Issue by super-iterator** [balderdashy/sails#3357]
I have a model like this:
```
userId : {
          model : 'user',
          required : true,
          unique : true,
}
```

From another place in the code, I call this
```
var ObjectId = require('mongodb').ObjectID;

MyModel.find( {userId:ObjectId( user.id ) } , function  (err,model) {

						console.log(model);
}
```
returns all the elements, and not considering the added criteria "{userId:user.id}"

It only works when I change the model type to string, like this:

```
userId : {
          type        : 'string',
          required  : true,
          unique    : true,
}
```

**Comment by sailsbot** [comment]
Thanks for posting, @super-iterator.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
